node/src/neighborhood/mod.rs

```diff
@@ -1948,12 +2095,81 @@ mod tests {
 
         let assertions = Box::new(move |actor: &mut Neighborhood| {
             assert_eq!(
-                actor.overall_connection_status.progress,
-                vec![ConnectionProgress {
-                    initial_node_descriptor: node_descriptor.clone(),
-                    current_peer_addr: node_ip_addr,
-                    connection_stage: ConnectionStage::Failed(NoGossipResponseReceived)
-                }]
+                actor.overall_connection_status,
+                OverallConnectionStatus {
+                    can_make_routes: false,
+                    stage: OverallConnectionStage::NotConnected,
+                    progress: vec![ConnectionProgress {
+                        initial_node_descriptor: node_descriptor.clone(),
+                        current_peer_addr: node_ip_addr,
+                        connection_stage: ConnectionStage::Failed(NoGossipResponseReceived)
+                    }]
+                }
+            );
+        });
+        addr.try_send(AssertionsMessage { assertions }).unwrap();
+        System::current().stop();
+        assert_eq!(system.run(), 0);
+    }
+
+    #[test]
+    pub fn progress_in_the_stage_of_overall_connection_status_made_by_one_cpm_is_not_overriden_by_the_other(
+    ) {
+        init_test_logging();
+        let peer_1 = make_ip(1);
+        let peer_2 = make_ip(2);
+        let initial_node_descriptors =
+            vec![make_node_descriptor(peer_1), make_node_descriptor(peer_2)];
+        let neighborhood_config = NeighborhoodConfig {
+            mode: NeighborhoodMode::Standard(
+                NodeAddr::new(&make_ip(3), &[1234]),
+                initial_node_descriptors,
+                rate_pack(100),
+            ),
+        };
+        let mut subject = Neighborhood::new(
+            main_cryptde(),
+            &bc_from_nc_plus(
+                neighborhood_config,
+                make_wallet("earning"),
+                None,
+                "progress_in_the_stage_of_overall_connection_status_made_by_one_cpm_is_not_overriden_by_the_other"),
+        );
+        let (node_to_ui_recipient, _) = make_node_to_ui_recipient();
+        subject.node_to_ui_recipient_opt = Some(node_to_ui_recipient);
+        let addr = subject.start();
+        let cpm_recipient = addr.clone().recipient();
+        let system = System::new("testing");
+        cpm_recipient
+            .try_send(ConnectionProgressMessage {
+                peer_addr: peer_1,
+                event: ConnectionProgressEvent::TcpConnectionSuccessful,
+            })
+            .unwrap();
+        cpm_recipient
+            .try_send(ConnectionProgressMessage {
+                peer_addr: peer_1,
+                event: ConnectionProgressEvent::IntroductionGossipReceived(make_ip(4)),
```

Comment:
A short comment would make it much more clearer. Maybe: Now the stage is ConnectedToNeighbor.